crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -1932,8 +1916,6 @@ impl IndexerStore for PgIndexerStore {
             self.metrics.last_pruned_checkpoint.set(cp as i64);
         }
 
-        // NOTE: prune epochs table last, otherwise get_checkpoint_range_for_epoch would fail.
-        self.prune_epochs_table(epoch).await?;
```

Comment:
We do `SELECT min(epoch) FROM epochs` to [find the minimum prunable epoch](https://github.com/MystenLabs/sui/blob/f0591bdfb5ebc6db8804e3d6cd9da49b920577dd/crates/sui-indexer/src/store/pg_indexer_store.rs#L224) and start pruning if this value is less than (current epoch - epochs to keep). With epochs tables being never pruned, that means this min epoch will always be 0 and we will be [trying to drop partitions and prune tables starting from epoch 0 all the time](https://github.com/MystenLabs/sui/blob/f0591bdfb5ebc6db8804e3d6cd9da49b920577dd/crates/sui-indexer/src/handlers/pruner.rs#L87). I don't think this will create any correctness issues but want to double check that it is the intended behavior.

Reply:
I think https://github.com/MystenLabs/sui/pull/19318 would eliminate that issue.

Reply:
Ah no actually. It only makes the partition part not dependent on min epoch, but the table pruning part still does.
I will take a look. Good catch!

Reply:
nice catch indeed, the root cause is that `epochs` table serves as both a data source table and a pruning watermark table. When we have [#19368](https://github.com/MystenLabs/sui/pull/19368) in, we can get the min prunable range from the new `watermarks` table, with the nuance that we will need an extra column in `watermarks` marking if the table is prunable at all.

It will be cleaner after the separation, as instead of one holistic pruner, each table can prune without having one holistic starting point for pruning.

Reply:
adding @wlmyng to this thread

Reply:
https://github.com/MystenLabs/sui/pull/19535 should have resolved this issue now.

Reply:
I see after #19535 we prune starting from `prune_start_epoch` instead of `min_epoch`. But in the very first iteration, `next_prune_epoch` is `None` so we will still start from `min_epoch` which is 0?

Reply:
Yes and I think that's ok. The cost will just be two DB queries per checkpoint, since we call `get_min_prunable_checkpoint` before starting to prune the epoch, which leverages the `pruner_cp_watermark` table.